### PR TITLE
Add API v1 endpoints

### PR DESCRIPTION
**Pull request by rez-gitlab**
## Description
This PR adds API endpoints.

## Changes
- Added new service class
- Updated routes
- Added tests

## Testing
- [ ] Unit tests pass
- [ ] Integration tests pass

## Screenshots
_Add screenshots here for UI changes_

### CODE REVIEW

**Comment by rez-gitlab** [comment]
LGTM! Nice implementation 👍

**Comment by rez-gitlab** [comment]
Could we add some tests for edge cases?

```ruby
it 'handles nil input' do
  expect(subject.call(nil)).to be_nil
end
```

**Comment by rez-gitlab** [comment]
I've addressed the feedback. Ready for re-review.

**Comment by rez-gitlab** [comment]
Screenshot 2025-11-25 at 13 04 03
here is some screenshot

**Comment by georgekoltsov** [comment]
image
image
image

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class UsersController < ApplicationController
```
Comment (georgekoltsov) [inline app/controllers/api/v1/users_controller.rb:5]: imageimageimageimageimageimageimageimageimageimageimageimageimage

**Review by georgekoltsov — CHANGES_REQUESTED** [review]
